src/browser/connection/index.ts

```diff
@@ -205,18 +205,23 @@ export default class BrowserConnection extends EventEmitter {
         this.openFileProtocolUrl = proxy.resolveRelativeServiceUrl(this.openFileProtocolRelativeUrl);
     }
 
-    public set messageBus (messageBus: MessageBus | undefined) {
-        this._messageBus         = messageBus;
+    public initMessageBusEvents (): void {
         this.warningLog.callback = WarningLog.createAddWarningCallback(this._messageBus);
 
-        if (messageBus) {
-            messageBus.on('test-run-start', testRun => {
+        if (this._messageBus) {
+            this._messageBus.on('test-run-start', testRun => {
                 if (testRun.browserConnection.id === this.id)
                     this._currentTestRun = testRun;
             });
         }
 
-        this.emit('message-bus-initialized', messageBus);
+        this.emit('message-bus-initialized', this._messageBus);
+    }
+
+    public set messageBus (messageBus: MessageBus | undefined) {
```

Comment:
It's a bad design.
Property setter should contain the event intialization. Split this.